### ISSUE TITLE
RouterSolicit with no options generates an error

### ISSUE THREAD

**Issue by leshow** [libpnet/libpnet#583]
A pcap from my network shows that router solicitations are sent with no options.

![solicit_noopts](https://user-images.githubusercontent.com/1128302/188219922-61b6a5a2-57ee-42f9-b5c9-3de6b7669c69.png)

If I construct a RouterSolicit packet with the pnet lib however, leaving the options field empty leads to an invalid message.

![solicit_opts](https://user-images.githubusercontent.com/1128302/188220093-325fe29c-87dd-4db8-b228-d47bf4d577e0.png)

This message is constructed with:
```
 let adv = ndp::RouterSolicit {
            icmpv6_type: Icmpv6Types::RouterSolicit,
            icmpv6_code: Icmpv6Code(0),
            checksum: 0,
            options: Vec::new(),
            reserved: 0,
            payload: Vec::new(),
};
```

wireshark shows it as an error

**Comment by stappersg** [comment]
And what should be the outcome of this issue?

**Comment by leshow** [comment]
Right, sorry. The options section is a variable length block at the end of the packet, in the good case if there are no options it is not present. In the `pnet` case it exists with a type of `Unknown(0)` and then length 0.

Router solicitations sometimes have a link-layer address included in the options section, but according to the RFC this is not always the case. 

 >    Valid Options:
 >     Source link-layer address The link-layer address of the sender, if
 >                    known.  MUST NOT be included if the Source Address
 >                   is the unspecified address.  Otherwise, it SHOULD
 >                   be included on link layers that have addresses.
https://www.rfc-editor.org/rfc/rfc4861#section-4.1

I would be open to fixing this but would need some direction, I have not touched the source yet.

**Comment by stappersg** [comment]
What I think:

* It is fine that `libpnet` allows to do _stupid_ things as sending empty packets.
* It is great that `libpnet` doesn't prevent doing _smart_ things.
* This issue will linger for a long time. (Please prove me wrong.)

**Comment by leshow** [comment]
I don't think the issue has anything to do with the points you listed. I'm not sure how to disprove "this issue will linger for a long time"... I've offered to fix it if someone can point me in the right direction, which you are welcome to do also if you know how to resolve it.

Let's take a step back, the issue is that `libpnet` creates an incorrect packet when the options field is empty. We should be able to create a router solicit with no data in the opts section, but instead an option *is* added with a type of 0 and a length of 0 (see the second screenshot). I don't think there is anything complicated going on here. It is just broken.
